devito/types/sparse.py

```diff
@@ -89,10 +156,89 @@ def space_order(self):
         """The space order."""
         return self._space_order
 
+    @property
+    def r(self):
```

Comment:
what I'm saying here applies everywhere, even to the new WeightedInterpolator class. 

Instance/class properties/methods should always be logically grouped together according to some logic. For example, first all of the private properties, then all of the public properties, then...

Here `r` is in between private stuff, some being methods and some properties

Reply:
up